tests/pds/dataservice_test.go

```diff
@@ -313,30 +316,9 @@ var _ = Describe("{ScaleUPDataServices}", func() {
 
 				Step("Running Workloads before scaling up of dataservices ", func() {
 					log.InfoD("Running Workloads on DataService %v ", ds.Name)
-					if ds.Name == postgresql {
-						deploymentName := "pgload"
-						log.Infof("Running Workloads on DataService %v ", ds.Name)
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds.Name, deployment.GetId(), "100", "1", deploymentName, namespace)
-						log.FailOnError(err, "Error while genearating workloads")
-					}
-					if ds.Name == rabbitmq {
-						deploymentName := "rmq"
-						log.Infof("Running Workloads on DataService %v ", ds.Name)
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds.Name, deployment.GetId(), "", "", deploymentName, namespace)
-						log.FailOnError(err, "Error while genearating workloads")
-					}
-					if ds.Name == redis {
-						deploymentName := "redisbench"
-						log.Infof("Running Workloads on DataService %v ", ds.Name)
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds.Name, deployment.GetId(), "", "", deploymentName, namespace)
-						log.FailOnError(err, "Error while genearating workloads")
-					}
-					if ds.Name == cassandra {
-						deploymentName := "cassandra-stress"
-						log.Infof("Running Workloads on DataService %v ", ds.Name)
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds.Name, deployment.GetId(), "", "", deploymentName, namespace)
-						log.FailOnError(err, "Error while genearating workloads")
-					}
+					var params pdslib.WorkloadGenerationParams
+					pod, dep, err = RunWorkloads(params, ds, deployment, namespace)
+					log.FailOnError(err, "Error while genearating workloads")
```

Comment:
Can we elaborate what workloads we failed to generate here? Something to give us a hint or more info about what app failed deployment
```
log.FailOnError(err, fmt.Sprintf("Error while genearating workloads [%s]", ds.Name))
```

Reply:
done